src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -239,6 +249,21 @@ public void putMetric(Metric<Double> metric, Dimensions dimensions, long windowS
     return create.select().from(DSL.table(metric)).fetch();
   }
 
+  public Result<Record> queryMetric(String metric, Collection<String> dimensions, int limit) {
+    if (!DBUtils.checkIfTableExists(create, metric)) {
+      return null;
+    }
+    if (limit < 0) {
+      throw new IllegalArgumentException("Limit must be non-negative");
```

Comment:
This is too late to validate if the limit is negative. Maybe we should do it up in the call handling stack ?

Reply:
Such a check is enforced multiple times in QueryBatchRequestHandler. The check is also included here in MetricsDB since adding this function expands the API of MetricsDB, and such a check is required for the new expanded API.